### ISSUE TITLE
log Basic Auth username in accesslog

### ISSUE THREAD

**Issue by bastiaanb** [containous/traefik#2060]
### Do you want to request a *feature* or report a *bug*?

Yes ;-)
I don't know whether the behaviour is already expected to work....

### What did you do?

* configured traefik to perform Basic Auth on the default entrypoint. 
* enabled access logging.
* started traefik:
```shell
docker run --rm -i -p 10080:80 -p 18080:8080 -v $(pwd)/traefik.toml:/etc/traefik/traefik.toml traefik:1.4.0-rc1
```
I performed a authenticated request with curl:
```shell
curl -H "Host: httpbin.org" --user test:test http://localhost:10080/headers
```

### What did you expect to see?
In the ClientUsername field of the access log I expect to see the user 'test' logged. E.g.
```shell
172.17.0.1 - test [05/Sep/2017:09:14:16 +0000] "GET /headers HTTP/1.1" 200 292 - "curl/7.51.0" 1 "frontend1" "http://httpbin.org:80" 241ms
```

### What did you see instead?
I see a dash instead:
```
172.17.0.1 - - [05/Sep/2017:09:14:16 +0000] "GET /headers HTTP/1.1" 200 292 - "curl/7.51.0" 1 "frontend1" "http://httpbin.org:80" 241ms
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0-rc1
Codename:     roquefort
Go version:   go1.9
Built:        2017-08-29_08:35:24AM
OS/Arch:      linux/amd64
```

Same results with Traefik 1.3.7

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

<details>
<summary>configuration</summary>

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "/dev/stdout",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "LogLevel": "ERROR",
 "EntryPoints": {
  "http": {
   "Network": "",
   "Address": ":80",
   "TLS": null,
   "Redirect": null,
   "Auth": {
    "Basic": {
     "Users": [
      "test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/",
      "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"
     ],
     "UsersFile": ""
    },
    "Digest": null,
    "Forward": null,
    "HeaderField": ""
   },
   "WhitelistSourceRange": null,
   "Compress": false,
   "ProxyProtocol": false
  }
 },
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [
  "http"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": {
  "ReadTimeout": 0,
  "WriteTimeout": 0,
  "IdleTimeout": 180000000000
 },
 "ForwardingTimeouts": {
  "DialTimeout": 30000000000,
  "ResponseHeaderTimeout": 0
 },
 "Docker": null,
 "File": {
  "Watch": true,
  "Filename": "",
  "Constraints": null,
  "Trace": false,
  "DebugLogGeneratedTemplate": false,
  "Directory": ""
 },
 "Web": {
  "Address": ":8080",
  "CertFile": "",
  "KeyFile": "",
  "ReadOnly": false,
  "Statistics": null,
  "Metrics": null,
  "Path": "",
  "Auth": null,
  "Debug": false,
  "CurrentConfigurations": null,
  "Stats": null,
  "StatsRecorder": null
 },
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": "/etc/traefik/traefik.toml"
}
```
</details>

<br/>

```toml
accesslogsfile = "/dev/stdout"
defaultEntryPoints = ["http"]
[accessLogs]
  format = "JSON"
  filepath = "/dev/stdout"
[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.http.auth.basic]
    users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]

[file]

[backends]
  [backends.backend1]
    [backends.backend1.servers.server1]
      url = "http://httpbin.org:80"

[frontends]
  [frontends.frontend1]
    backend = "backend1"
    passHostHeader = true
    [frontends.frontend1.routes.test_1]
      rule = "PathPrefix:/"

[web]
  address = ":8080"
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2017-09-05T09:17:15Z" level=debug msg="Basic auth success..." 
time="2017-09-05T09:17:15Z" level=debug msg="Round trip: http://httpbin.org:80, code: 200, duration: 233.084177ms" 
172.17.0.1 - - [05/Sep/2017:09:17:15 +0000] "GET /headers HTTP/1.1" 200 292 - "curl/7.51.0" 1 "frontend1" "http://httpbin.org:80" 233ms
```

**Comment by bastiaanb** [comment]
As dicussed on Slack:
One option to implement would be to set the username in req.URL after authentication.
Another would be to extend the logging code to obtain the username from both sources.

**Comment by bastiaanb** [comment]
I've tried to implement this by setting r.URL.User in the authenticator.handler, but apparently it does not get logged. 
Interestingly the same function sets X-Webauth-User (or what you have defined as auth headerField). This correclty gets passed to the backend server, but also ends up empty in the (JSON) log:
```
{
  "BackendAddr": "httpbin.org:80",
  "BackendName": "backend1",
  "BackendURL": {
    "Scheme": "http",
    "Opaque": "",
    "User": null,
    "Host": "httpbin.org:80",
    "Path": "",
    "RawPath": "",
    "ForceQuery": false,
    "RawQuery": "",
    "Fragment": ""
  },
  "ClientAddr": "[::1]:42980",
  "ClientHost": "::1",
  "ClientPort": "42980",
  "ClientUsername": "-",
  "DownstreamContentSize": 332,
  "DownstreamStatus": 200,
  "DownstreamStatusLine": "200 OK",
  "Duration": 233861988,
  "FrontendName": "frontend1",
  "OriginContentSize": 332,
  "OriginDuration": 233481094,
  "OriginStatus": 200,
  "OriginStatusLine": "200 OK",
  "Overhead": 380894,
  "RequestAddr": "httpbin.org",
  "RequestContentSize": 0,
  "RequestCount": 2,
  "RequestHost": "httpbin.org",
  "RequestLine": "GET /headers HTTP/1.1",
  "RequestMethod": "GET",
  "RequestPath": "/headers",
  "RequestPort": "-",
  "RequestProtocol": "HTTP/1.1",
  "RetryAttempts": 0,
  "StartLocal": "2017-09-05T15:02:21.621815783+02:00",
  "StartUTC": "2017-09-05T13:02:21.621815783Z",
  "downstream_Access-Control-Allow-Credentials": "true",
  "downstream_Access-Control-Allow-Origin": "*",
  "downstream_Content-Length": "332",
  "downstream_Content-Type": "application/json",
  "downstream_Date": "Tue, 05 Sep 2017 13:02:21 GMT",
  "downstream_Server": "meinheld/0.6.1",
  "downstream_Via": "1.1 vegur",
  "downstream_X-Powered-By": "Flask",
  "downstream_X-Processed-Time": "0.000715017318726",
  "level": "info",
  "msg": "",
  "origin_Access-Control-Allow-Credentials": "true",
  "origin_Access-Control-Allow-Origin": "*",
  "origin_Content-Length": "332",
  "origin_Content-Type": "application/json",
  "origin_Date": "Tue, 05 Sep 2017 13:02:21 GMT",
  "origin_Server": "meinheld/0.6.1",
  "origin_Via": "1.1 vegur",
  "origin_X-Powered-By": "Flask",
  "origin_X-Processed-Time": "0.000715017318726",
  "request_Accept": "*/*",
  "request_Authorization": "Basic dGVzdDp0ZXN0",
  "request_User-Agent": "curl/7.51.0",
  "request_X-WebAuth-User": "",
  "time": "2017-09-05T15:02:21+02:00"
}
```

**Comment by bastiaanb** [comment]
relevant commit https://github.com/bastiaanb/traefik/commit/e806d1b1f1aba54a5730448eff707737e3ec1600

**Comment by ldez** [comment]
Closed by #2111
